// angular-toastify/esm5/lib/toast.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable, EventEmitter } from '@angular/core';
import { Toast } from './toast';
import { ToastType } from './toast-type';
import * as i0 from "@angular/core";
var ToastService = /** @class */ (function () {
    function ToastService() {
        this.toastAddedEvent = new EventEmitter();
        this.dismissAllEvent = new EventEmitter();
    }
    /**
     * @return {?}
     */
    ToastService.prototype.dismissAll = /**
     * @return {?}
     */
    function () {
        this.dismissAllEvent.emit();
    };
    /**
     * @param {?} message
     * @return {?}
     */
    ToastService.prototype.info = /**
     * @param {?} message
     * @return {?}
     */
    function (message) {
        /** @type {?} */
        var toast = new Toast(message, ToastType.info);
        this.toastAddedEvent.emit(toast);
    };
    /**
     * @param {?} message
     * @return {?}
     */
    ToastService.prototype.success = /**
     * @param {?} message
     * @return {?}
     */
    function (message) {
        /** @type {?} */
        var toast = new Toast(message, ToastType.success);
        this.toastAddedEvent.emit(toast);
    };
    /**
     * @param {?} message
     * @return {?}
     */
    ToastService.prototype.warn = /**
     * @param {?} message
     * @return {?}
     */
    function (message) {
        /** @type {?} */
        var toast = new Toast(message, ToastType.warning);
        this.toastAddedEvent.emit(toast);
    };
    /**
     * @param {?} message
     * @return {?}
     */
    ToastService.prototype.error = /**
     * @param {?} message
     * @return {?}
     */
    function (message) {
        /** @type {?} */
        var toast = new Toast(message, ToastType.error);
        this.toastAddedEvent.emit(toast);
    };
    /**
     * @param {?} message
     * @return {?}
     */
    ToastService.prototype.default = /**
     * @param {?} message
     * @return {?}
     */
    function (message) {
        /** @type {?} */
        var toast = new Toast(message, ToastType.default);
        this.toastAddedEvent.emit(toast);
    };
    ToastService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    /** @nocollapse */
    ToastService.ctorParameters = function () { return []; };
    /** @nocollapse */ ToastService.ngInjectableDef = i0.ɵɵdefineInjectable({ factory: function ToastService_Factory() { return new ToastService(); }, token: ToastService, providedIn: "root" });
    return ToastService;
}());
export { ToastService };
if (false) {
    /** @type {?} */
    ToastService.prototype.toastAddedEvent;
    /** @type {?} */
    ToastService.prototype.dismissAllEvent;
}
